### PR TITLE
improves MarkDuplcates documentation, fixes #1087

### LINKED ISSUE #1087: MarkDuplicate documentation incorrect
### Affected tool(s)
_MarkDuplicates READ_NAME_REGEX_

### Affected version(s)
- [x ] Latest public release version 2.17.2
- [ x] Latest development/master branch as of 2018-01-18

### Description 
Mark dup docs say 
"Note that if the default regex is specified, a regex match is not actually done"...
the [default regex](https://github.com/broadinstitute/picard/blob/e7d6c8f8cf110e6f74edefdef86fba64d01ad0fa/src/main/java/picard/sam/util/ReadNameParser.java#L27) is "<optimized capture of last three ':' separated fields as numeric values>".intern(); 
it's not possible to specify a command line argument that matches this  (due to the intern())

I'll send a PR related to this to fix the documentation.

PS
I found this code confusing since DEFAULT_READ_NAME_REGEX is not actually a regex.

### PR DESCRIPTION
**Pull request by bwlang**
### Description
This updates the MarkDuplicates documentation to improve clarity and correct an inaccuracy.

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/sam/markduplicates/util/AbstractOpticalDuplicateFinderCommandLineProgram.java [hunk 1]
```diff
@@ -37,18 +37,22 @@
...
 
-    @Argument(doc = "Regular expression that can be used to parse read names in the incoming SAM file. Read names are " +
-            "parsed to extract three variables: tile/region, x coordinate and y coordinate. These values are used " +
-            "to estimate the rate of optical duplication in order to give a more accurate estimated library size. " +
-            "Set this option to null to disable optical duplicate detection, e.g. for RNA-seq " +
+    @Argument(doc = "MarkDuplicates can use the tile and cluster positions to estimate the rate of optical duplication " +
+            "in addition to the dominant source of duplication, PCR, to provide a more accurate estimation of library size. " +
+            "By default (with no READ_NAME_REGEX specified), MarkDuplicates will attempt to extract coordinates " +
+            "using a split on ':' (see Note below).  " +
+            "Set READ_NAME_REGES to "null" to disable optical duplicate detection, e.g. for RNA-seq " +
```
Comment (yfarjoun) [inline src/main/java/picard/sam/markduplicates/util/AbstractOpticalDuplicateFinderCommandLineProgram.java]: READ_NAME_REGES-> READ_NAME_REGEX

**Comment by gbggrant** [comment]
@sooheelee bump

**Comment by gbggrant** [comment]
@sooheelee does this look ok?

**Comment by bwlang** [comment]
@tfenne  : do you care to comment on this?

**Comment by yfarjoun** [comment]
this is fine.

**Pull request merged**